doc/source/extended_documentation/wxcode/build_a_decision_tree.rst

```diff
@@ -0,0 +1,111 @@
+**Weather symbol decision trees**
+
+Weather symbol decision trees use diagnostic fields to diagnose a suitable
+symbol to represent the weather conditions. The tree is comprised of a series
+of interconnected decision nodes. At each node one or multiple forecast
+diagnostics are compared to predefined threshold values. The node has a success
+and failure path on to the next node, or on to a resulting weather symbol. By
+traversing the nodes it should be possible, given the right weather conditions,
+to arrive at any of the weather symbols.
+
+The first few nodes of a decision tree are represented in the schematic below.
+
+.. figure:: extended_documentation/wxcode/thunder_nodes.png
+     :align: center
+     :scale: 80 %
+     :alt: Schematic of thundery nodes in a decision tree
+
+There are two thresholds being used in these nodes. The first is the diagnostic
+threshold which identifies the critical value for a given diagnostic. In the
+first node of the schematic shown this threshold is the count of lightning
+flashes in an hour exceeding 0.0. The second threshold is the probability of
+exceeding (in this case) this diagnostic threshold. In this first node it's a
+probability of 0.3 (30%). So the node overall states that if there is an equal
+or greater than 30% probability of any lightning flashes in the hour being
+forecast, proceed to the succeed node, else move to the fail node.
+
+**Encoding a decision tree**
+
+The first node above is encoded as follows::
+
+  {
+    "lightning": {
+        "succeed": "lightning_cloud",
+        "fail": "heavy_precipitation",
+        "diagnostic_missing_action": "fail",
+        "probability_thresholds": [0.3],
+        "threshold_condition": ">=",
+        "condition_combination": "",
+        "diagnostic_fields": [
+            "probability_of_number_of_lightning_flashes_per_unit_area_in_vicinity_above_threshold"
+        ],
+        "diagnostic_thresholds": [[0.0, "m-2"]],
+        "diagnostic_conditions": ["above"]
+    },
+  }
+
+The key at the first level, "lightning" in this case, names the node so that it
+can be targeted as a succeed or fail destination from other nodes. The dictionary
+accessed with this key contains the essentials that make the node function.
+
+  - **succeed**: The next node to test if the tests in this node succeed.
```

Comment:
@MoseleyS I don't object, but could we do this after the simplification to a single tree? It would mean less things to modify as there will be only one json to pull in, and only one stored for the unit tests. I've also got a branch where I had modified the tree to use binary/mask fields (for the "is it showery" field) which I need to put on top of this one, so adding more changes now will make that harder.

So my suggestion is I take you proposal and stick it in a ticket and add it to this epic, so I can do it as a tidy up exercise at the end.

Reply:
Yes, that's reasonable. I'd still like some of this documentation improved though - can you consider using some of the changed sentences while deferring the proposed key names?

Reply:
Done, and a ticket created to record the key changes: https://github.com/metoppv/mo-blue-team/issues/46